specs/capri/Passwor_Encryption_Design.md

```diff
@@ -0,0 +1,117 @@
+# Title
+
+**Author(s)**: [Jack Li](https://github.com/lasshpanzer)
+
+## Summary
+
+This document is a design proposal to provide a password security mechanism for OpenSDS.
+
+## Motivation
+
+Currently, the password used by other components such as cinder, kubernetes, MultiCloud to interact with OpenSDS is not encrypted, which means the user’s information can be leaked easily.
+
+### Goals
+
+The password shall be encrypted when other components use it to interact with OpenSDS.
+
+1.A default encryption tool will be provided for users to obtain the cipher text.
+
+2.Which encryption tool is selected and whether the password is encrypted can be configurable. By default, the encryption tool is AES which is an open source tool, and the password is not encrypted.
+
+
+## Design Details
+
+1.Add options to the OpenSDS configuration file that enable user to choose the password encrypter tool they want and choose whether to encrypt the password. The encryption tools can be implemented by the user. These two options added to the opensds.conf are as follows:
+
+// Whether to encrypt the password. If enabled, the value of the password must be cipher text.
+enable_encrypted = False
+// Encryption and decryption tool. Default value is aes. The decryption tool can only decrypt the corresponding ciphertext.
+pwd_encrypter = aes
+
```

Comment:
Add "\`\`\`" on the line before 26.  Add "\`\`\`" on the line after "pwd_encrypter":

\`\`\`
// Whether to encrypt the password. If enabled, the value of the password must be cipher text.
enable_encrypted = False
// Encryption and decryption tool. Default value is aes. The decryption tool can only decrypt the corresponding ciphertext.
pwd_encrypter = aes
\`\`\`